### ISSUE TITLE
Executing commands on sessions without attaching

### ISSUE THREAD

**Issue by mMontu** [tmuxinator/tmuxinator#376]
Is it possible to run a command on a project from a shell outside tmux and capture its output? Something similar to `tmux -send-keys`, but retrieving the output of the commands.

Context: I often need to run commands on some remote servers, thus I created some scripts to run those commands through ssh. The problem is that some commands requires (the same) long setup to be issued. It would be far better to run the setup once.

Using tmuxinator projects is great because `mux remoteProject` will run the setup on the first execution, and just attach to the existing tmux session on subsequent calls. I have been playing with passing arguments to the project (containing the command to be executed) and `post:` (to execute the project argument, capture the output and exit tmux) but without success.

**Comment by adamstrickland** [comment]
I'm far from a definitive source, but IMO this functionality would be getting away from the Unix philosophy of [do one thing and do it well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well).  

It's a little fuzzy to me without a concrete example of what you're trying to do, but on the surface it sounds like some custom scripts (perhaps co-located next to your project file, so they can easily find it) might be a better solution.

**Comment by mMontu** [comment]
I'm an adept of DOTADIW as well, and I thought that this feature would fall in the "one thing":

>  Create and manage tmux sessions easily.

The idea was to use tmuxinator to create sessions e access existing sessions instead of creating duplicates, and them sending commands/receiving output non-interactive (this might be the part that doesn't follow the DOTADIW...).

The actual use case is to perform development on legacy systems. One example is issuing ClearCase commands on a remote server through ssh; currently I used something like this:

    ssh <remote_host> "cleartool setview -login -exec '%s' <cc_view>" 

The `%s` contains the actual command. This approach works fine, but as I increased the complexity I started noticing a delay when issuing several commands. After some analysis it became clear that setting the view is the bottleneck. As I already use tmuxinator on a daily basis (and I'm very happy with it :)) I thought it would be nice to use it through scripts, to have something beyond [reuse of ssh connections](http://rawsyntax.com/blog/faster-ssh-connections/), i.e., reuse the entire enviroment.
